sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -207,6 +207,29 @@ trait CheckAnalysis extends PredicateHelper {
                 s"of type ${condition.dataType.catalogString} is not a boolean.")
 
           case Aggregate(groupingExprs, aggregateExprs, child) =>
+            val aggExpressions = aggregateExprs.flatMap { e =>
+              e.collect {
+                case ae: AggregateExpression => ae
+              }
+            }
+            val aggWithDistinctAndFilters = aggExpressions.filter(e => e.isDistinct)
```

Comment:
Why is this `aggWithDistinctAndFilters` since this is only checking for distinct?

Reply:
Thanks for your review.
This PR only support multi DISTINCT aggregate expression goes with FILTER using the same attributes.
If DISTINCT aggregate expression contains different attributes, let's throw an analysis exception.